src/EditorFeatures/CSharpTest/ImplementInterface/ImplementInterfaceTests.cs

```diff
@@ -8181,52 +8181,174 @@ public class Test : ITest
         }
 
         [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsImplementInterface)]
-        public async Task GenericInterfaceNotNull1()
+        public async Task TestAllowNullAttributeOnNonNullableReferenceType()
         {
-            await TestInRegularAndScriptAsync(
-@$"#nullable enable 
+            await TestInRegularAndScript1Async(
+$@"#nullable enable
 
 using System.Diagnostics.CodeAnalysis;
 
 {NullableAttributesCode}
 
-interface IFoo<T>
+interface IInterface<T>
 {{
-    [return: NotNull]
-    T Bar([DisallowNull] T bar);
+    [return: MaybeNull]
+    T Method([AllowNull] T value);
+}}
+
+class Class : [|IInterface<string>|]
+{{
+}}",
+$@"#nullable enable
+
+using System.Diagnostics.CodeAnalysis;
+
+{NullableAttributesCode}
+
+interface IInterface<T>
+{{
+    [return: MaybeNull]
+    T Method([AllowNull] T value);
+}}
+
+class Class : IInterface<string>
+{{
+    public string Method(string? value)
```

Comment:
Shouldn't this be `string?`, since the return was marked as [MaybeNull]?

Reply:
Same as above, see https://github.com/dotnet/roslyn/pull/38203#discussion_r336684600.

Reply:
Fixed.